### PR TITLE
Moved unlockProvider back into unlock-app

### PR DESCRIPTION
**Pull request by julien51**
# Description

This required surprising little changes beyond paths
I had to install a package (`nock`) for tests (maybe a follow up PR to see if we can remove that dependency...) and did a few cosmetic changes/renames.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
strangely, tests pass when I do `npm run test src/__tests__/services/unlockProvider.test.js` so they must be conflicting with something else.

File: unlock-app/src/__tests__/services/unlockProvider.test.js [hunk 1]
```diff
@@ -1,40 +1,196 @@
...
+
+        expect(
+          sigUtil.recoverPersonalSignature({
+            data: messageHash,
+            sig,
+          })
+        ).toEqual(publicKey.toLowerCase())
+      })
+
+      it('should be compatible with walletService', async done => {
```
Comment (cnasc) [inline unlock-app/src/__tests__/services/unlockProvider.test.js]: Weird that this test failed

Comment (julien51) [reply unlock-app/src/__tests__/services/unlockProvider.test.js]: Probably some network call. trying to see why.

Comment (julien51) [reply unlock-app/src/__tests__/services/unlockProvider.test.js]: What's weirder is that it passes locally.

Comment (julien51) [reply unlock-app/src/__tests__/services/unlockProvider.test.js]: I found it!

Comment (cnasc) [reply unlock-app/src/__tests__/services/unlockProvider.test.js]: What was the problem?

Comment (julien51) [reply unlock-app/src/__tests__/services/unlockProvider.test.js]: We are making a network call to get the network id! 

But apparently there is smething else.

**Pull request merged**